src/utils/helpers/UserEventHelpers.ts

```diff
@@ -0,0 +1,125 @@
+import { event, select } from 'd3-selection';
+import { DependencyNode } from '../../components/types';
+import {
+    centerScreenToDimension,
+    changeZoom,
+    findGroupBackgroundDimension,
+    findMaxDependencyLevel,
+    getHighLightedLabelColor,
+    hideHighlightBackground,
+    highlight,
+    LevelStorage,
+    selectAllNodes,
+    selectHighLightedNodes,
+    zoomToHighLightedNodes,
+} from './GraphHelpers';
+import { LabelColors, MAXIMUM_ZOOM_SCALE, MINIMUM_ZOOM_SCALE, Selectors, TextColors } from '../AppConsts';
+import { zoom, zoomIdentity } from 'd3-zoom';
+
+enum Subscriptions {
+    HIGHLIGHT = 'click.highlight',
+    RESET_HIGHLIGHT = 'click.resetHighlight',
+    CHANGE_HIGHLIGHT_RANGE = 'keydown.changeHighlightRange',
+    ZOOM_ON_ARROW_KEY = 'keydown.zoom',
+}
+
+export function subscribeToHighlight() {
+    selectAllNodes().on(Subscriptions.HIGHLIGHT, (node: DependencyNode) => {
+        LevelStorage.reset();
+        if (node.links.length) {
+            highlight(node);
+            zoomToHighLightedNodes();
+        }
+        event.stopPropagation();
+    });
+}
+
+export function subscribeToChangeHighlightRangeOnArrowKey() {
+    select('body').on(Subscriptions.CHANGE_HIGHLIGHT_RANGE, () => {
+        const labelNodesGroup = select<SVGGElement, DependencyNode>('g#labels');
+        LevelStorage.setMaxLevel(findMaxDependencyLevel(labelNodesGroup));
+
+        if (!isFinite(LevelStorage.getMaxLevel())) {
```

Comment:
maxLevel can be set to Infinity ?

Reply:
I have had a quick peek whether it can be infinite and haven't found any piece of code that would set maxLevel to infinity.